---
title: "coral_turf_contact_exp"
author: "Noam Altman-Kurosaki"
date: "2024-09-23"
output: html_document
---

```{r setup, include=FALSE}

rm(list=ls()) # clean up
library(here)
library(knitr)
library(plyr)
library(car)
library(reshape)
library(reshape2)
library(tidyr)
library(effects)
library(MASS)
library(MuMIn)
library(mgcv)
library(mgcViz)
library(sp)
library(lattice)
library(pbkrtest)
library(vegan)
library(ggplot2)
library(ggtext)
library(gridExtra)
library(grid)
library(tibble)
library(ggrepel)
library(rcompanion)
library(emmeans)
library(multcomp)
library(betareg)
library(glmmTMB)
library(DHARMa)
library(devtools)
library(tidyverse)
library(phyloseq)   
library(DESeq2)
library(microbiome)
library(picante)    
library(ALDEx2)
library(metagenomeSeq)  
library(HMP) 
library(dendextend)    
library(selbal) 
library(rms)
library(breakaway)
library(qiime2R)
# issues with unifrac if using phyloseq
# See: https://github.com/joey711/phyloseq/issues/956
# Using rbiom package for calculating unifrac distances instead
library(rbiom)
library(ggpubr)

opts_chunk$set(comment="  ",
               collapse=TRUE, 
               echo=FALSE,
               #fig.asp=1/gr,
               fig.height=8,
               fig.width = 10,
               dev="png",
               warning=TRUE
               )
opts_knit$set(eval.after = "fig.cap")
```  

```{r functions}
# create function for standard error
se = function(x){
  sd(x, na.rm = TRUE)/sqrt(length(x))
}


# function for ellipsess
#taken from the excellent stackoverflow Q+A: http://stackoverflow.com/questions/13794419/plotting-ordiellipse-function-from-vegan-package-onto-nmds-plot-created-in-ggplo
veganCovEllipse <- function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}




```

# Table S1

```{r turf species list, eval = FALSE}
turf <- read.csv(here::here("data", "turf_species_list_wide.csv"), stringsAsFactors = T)
turf[is.na(turf)] <- 0 # set empty cells to 0
turf$Comb.trt <- as.factor(paste(turf$Damselfish.trt, "-", turf$Month.trt)) # create new factor for aggregating
turf_list <- ddply(turf, "Comb.trt", numcolwise(sum)) # sum across cob trt to get number of samples containing species of algae
turf_list[,2:44] <- turf_list[,2:44]/12 # convert turf list to show proportion of samples it was present in per stratum
turf_list <- as.data.frame(t(turf_list)) # transpose and convert to dataframe
colnames(turf_list) <- turf_list[1,] # set colnames to be the first row of the data (factor)
turf_list <- turf_list[-1,] # remove first row (repetitive to column names due to t() fxn)

# write turf_list to directory of your choice for creation of table S1 in ESM
```

# Figure S1

```{r turf richness, eval = FALSE}
turf_species <- turf[,unlist(lapply(turf, is.numeric))] # subset just turf species based on if the column is a numeric

turf$Richness <- rowSums(turf_species) # species richness for a sample found by summing across all species present in that sample
hist(turf$Richness)
richnessglm <- glmmTMB(Richness ~ Damselfish.trt*Month.trt, data = turf, family = compois) # conway maxwell poisson for underdispersion

plot(simulateResiduals(richnessglm)) # residuals look good
summary(richnessglm)
Anova(richnessglm) # No difference in richness based on damselfish (X2 = 1.1, P = 0.3), month (X2 = 0.25, P = 0.6), or their interaction (X2 = 0.70, P = 0.4)
richness_means <- ddply(turf, c("Damselfish.trt", "Month.trt"), numcolwise(mean))
richness_se <- ddply(turf, c("Damselfish.trt", "Month.trt"), numcolwise(se))
# SN-M: 11.3 ± 0.60 
# SN-S: 10 ± 0.72
# SP-M: 9.5 ± 0.45
# SP-S: 9.8 ± 0.41

ggplot()+
  geom_boxplot(data = turf, aes(x = Damselfish.trt, y = Richness, colour = Month.trt), position = position_dodge(0.9),
               outlier.shape = NA, size = 1.1) +
  geom_point(data = turf, aes(x = Damselfish.trt, y = Richness, colour = Month.trt),
             size = 3, position = position_jitterdodge(dodge.width = 0.9, jitter.width = 0.3)) +
  geom_text(aes(x = 2.5, y = 15, label = "Damselfish: P = 0.29\nMonth: P = 0.61\nInt: P = 0.40"), hjust = 1, size = 10) +
    theme_classic() + 
  labs(y = "Species richness\n", x = "") +
  guides(colour=guide_legend(title="Month")) +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        legend.position = "bottom",
        legend.key.size = unit(2, 'cm'),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        axis.line = element_line(colour = "black"),
        axis.text.x = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain")
  )
  

```

# Table S2

```{r turf relative abundance}
turf_cover <- read.csv(here::here("data", "turf_species_cover.csv"), stringsAsFactors = T)
turf_cover[is.na(turf_cover)] <- 0 # set empty cells to 0
turf_cover <- ddply(turf_cover, c("Damselfish.trt", "Month.trt", "Sample.ID"), numcolwise(mean) ) # average across 5 fields of view to get relative abundance per sample


turf_cover_table <- ddply(turf_cover, c("Damselfish.trt", "Month.trt"), numcolwise(mean) ) # get mean cover
turf_cover_table <- t(turf_cover_table) # transpose into long table

turf_cover_table_se <- ddply(turf_cover, c("Damselfish.trt", "Month.trt"), numcolwise(se) ) # repeat with standard errors
turf_cover_table_se <- t(turf_cover_table_se)

# write both turf_cover_tables to csv and combine in word to create table S2
```

# Figure 1

## Turf permanova and ordination setup
```{r turf permanova and ordination}
turf_m <- data.matrix(turf_cover[,-c(1:4)]) # create matrix of just species
turf_m <- turf_m/100 # convert percentage data to proportional data

rownames(turf_m) <- turf_cover$Sample.ID # set sample ID as row name
turf_m_trans <- asin(sqrt(turf_m)) # create matrix with arcsin sqrt transformed data

# PERMANOVA
vegan::adonis2(vegdist(turf_m_trans, method = "bray") ~ Damselfish.trt*Month.trt, data = turf_cover) 
# Significant difference in turf communities based on damselfish (F = 59.1, P < 0.001), month (F - 5.5, P = 0.005), and interaction (F = 3.5, P = 0.02)

# assess if assumption of multivariate homogeneity of variance is violated
vegan::permutest(vegan::betadisper(vegdist(turf_m_trans, method = "bray"), as.factor(paste(turf_cover$Damselfish.trt, "-", turf_cover$Month.trt)))) # P = 0.26 

# set.seed(1032)
# ord2 <- metaMDS(turf_m_trans, dist	= "bray",	trymax	= 100,	k	= 2)
# ord2 # stress = 0.11

set.seed(1032)
ord3 <- metaMDS(turf_m_trans, dist	= "bray",	trymax	= 100,	k	= 3)
ord3 # stress = 0.08

#ord4 <- metaMDS(turf_m_trans, dist	= "bray",	trymax	= 100,	k	= 4)
#ord4 # stress  = 0.06

# 3 dimensional ordination cause notable reduction in stress while still being relatively low dimension


```

## plot
```{r turf plotting, fig.height=20, fig.width=20}

cTurf.site.scores <- as.data.frame(scores(ord3, "sites"))  #Using the scores function from vegan to extract the site scores and convert to a data.frame
cTurf.site.scores$Sample <- rownames(cTurf.site.scores)  # create a column of site names, from the rownames of data.scores
cTurf.site.scores$Treatment <- as.factor(paste(turf_cover$Damselfish.trt, "-", turf_cover$Month.trt))  #  add the treatment as a grouping vactor
cTurf.site.scores$Damselfish <- turf_cover$Damselfish.trt
cTurf.site.scores$Month <- turf_cover$Month.trt

cTurf.species.scores <- as.data.frame(scores(ord3, "species"))  #Using the scores function from vegan to extract the species scores and convert to a data.frame
cTurf.species.scores$species <- rownames(cTurf.species.scores)  # create a column of species, from the rownames of species.scores

df_ell.cTurf.Treatment <- data.frame() #sets up a data frame before running the function.
# extract ellipses
for(g in levels(cTurf.site.scores$Treatment)){
  df_ell.cTurf.Treatment <- rbind(df_ell.cTurf.Treatment, cbind(as.data.frame(with(cTurf.site.scores[cTurf.site.scores$Treatment==g,],                                                 veganCovEllipse(cov.wt(cbind(NMDS1,NMDS2),wt=rep(1/length(NMDS1),
                               length(NMDS1)))$cov,center=c(mean(NMDS1),mean(NMDS2)))))
                                      ,Treatment=g))
}
# bootleg way to add in factors for treatments
df_ell.cTurf.Treatment$Damselfish <- as.factor( c(rep("Stegastes nigricans", 202), rep("Stegastes punctatus", 202)) )
df_ell.cTurf.Treatment$Month <- as.factor(c(rep("March", 101), rep("September", 101),
                                               rep("March", 101), rep("September", 101)))

# data for labelling the ellipse
NMDS.mean <- aggregate(cTurf.site.scores[,c("NMDS1", "NMDS2")], 
                    list(group = cTurf.site.scores$Treatment), mean)

# plot
ggplot() + 
  # add the point markers
  geom_point(data=cTurf.site.scores,aes(x=NMDS1,y=NMDS2, colour=Damselfish, shape = Month),size=12, stroke =3) + 
  
  # add ellipse
  geom_path(data = df_ell.cTurf.Treatment, aes(x = NMDS1, y = NMDS2, col = Damselfish, linetype = Month), size = 4)+ 
  
  # add species labels
  geom_text_repel(data = cTurf.species.scores, aes(x = NMDS1, y = NMDS2, label = species),colour = "grey25", size = 10) +
  
  # add stats
  geom_text(aes(x = -1.3, y = 0.85, label = "Stress = 0.08\nSpecies: P < 0.001 \nMonth: P = 0.005 \nSpecies X Month: P = 0.02"), size = 13, hjust = 0) +
  geom_hline(yintercept = 0, linetype = 2, alpha = 0.3) +
  geom_vline(xintercept = 0, linetype = 2, alpha = 0.3) +
  scale_colour_manual(values = c("Stegastes nigricans" = "navy",
                                 "Stegastes punctatus" = "dodgerblue"),
                      label = c(expression(italic("S. nigricans")),
                               expression(italic("S. punctatus")) )) +
  coord_equal() +
  theme_bw() + 
  theme(axis.title.x = element_text(size=26), 
        axis.title.y = element_text(size=26),
        axis.text.x = element_text(size=26), 
        axis.text.y = element_text(size=26),
        legend.key.width=unit(5,"cm"),
        panel.background = element_blank(), 
        panel.grid.major = element_blank(),  #remove major-grid labels
        panel.grid.minor = element_blank(),  #remove minor-grid labels
        plot.background = element_blank(),
        legend.key.size = unit(1.5, "cm"), # increase legend size
        legend.text = element_text(size = 34), # increase legend text
        legend.title = element_text(size = 36),
        legend.position = "bottom", # move legend to top left
        legend.box = 'horizontal', # make legend horizontal
        legend.box.background = element_rect(colour = "black")) + # add box around legend
  guides(colour=guide_legend(title.position="top", # make color legend title go above legend
                                     title.hjust =0.5, nrow = 2),
         shape=guide_legend(title.position="top",  # make shape legend title go above legend
                                     title.hjust =0.5,
                            nrow = 2),
         path = guide_legend(title.position="top",  # make shape legend title go above legend
                                     title.hjust =0.5)
         ) +

  scale_shape_manual(values = c(16, 1) )
  


```

# Figure 2 - coral-algal contact experiment

## setup and analysis
```{r read in data and preliminary viz}
comb <- read.csv(here::here("data", "coral_algal_contact_data.csv"), stringsAsFactors = T) # read data
comb$Colony.ID <- as.factor(paste(comb$Coral.treatment, comb$Block)) # add colony ID factor for random effect





```

## week 1 statistics - not used in final MS

```{r week 1 stats}
week1 <- subset(comb, Week == 1) # subset initial week 1 data
# Fit beta mixed model
beta_week1mme <- glmmTMB(Yield ~ Algal.treatment*Coral.treatment + (1|Colony.ID), data = week1, beta_family(link = "logit"))

# check model diagnostics with DHARMa
plot(week1_diag <- simulateResiduals(beta_week1mme))

summary(beta_week1mme) # low var in random effects (std = 0.01)
Anova(beta_week1mme) # sig algal trt effect (X2 = 88.2, P < 2E-16) and interaction (X2 = 13.5, P = 0.03), but no difference b/w coral trt (X2 = 0.13, P = 0.7)
emmeans(beta_week1mme, pairwise ~ Coral.treatment | Algal.treatment, adjust = "bonferroni")
# interaction driven by lower P. rus yield following 6d of contact with S. punctatus turf 


emmeans(beta_week1mme, pairwise ~ Algal.treatment | Coral.treatment, adjust = "bonferroni")
# A pulchra - the two turfs are different from turf mimic and Amansia.
# Porites - two turfs lower than empty, mimics, and sarg. S punctatus lower than Amansia as well.



```


## week 2 analyses - USED FOR FINAL MS

```{r week 2}

week2 <- subset(comb, Week == 2)

beta_week2mme <- glmmTMB(Yield ~ Algal.treatment*Coral.treatment + (1|Colony.ID), data = week2, beta_family(link = "logit"))
plot(week2_diag <- simulateResiduals(beta_week2mme)) 
summary(beta_week2mme) # low var in random effects (std = 0.04)
Anova(beta_week2mme)
# Algal treatment (X2 = 257, P < E-16)
# Coral trt (X2 = 6.4, P = 0.01)
# Int (X2 = 24.3, P = 4.6E-04)
r.squaredGLMM(beta_week2mme) # 0.86, 0.87
emmeans(beta_week2mme, ~ Coral.treatment | Algal.treatment, type = "response")
# within factor pairwise comparisons - differences between how algae affects corals
emmeans(beta_week2mme, pairwise ~ Coral.treatment | Algal.treatment, adjust = "bonferroni")
# P rus significantly impacted by both turfs more than A pulchra
# S. nigricans (t = 2.9, P = 0.004), S punctatus (t = 4.5, P < 0.001)

# # within factor pairwise comparisons - differences between how algal treatments within corals
emmeans(beta_week2mme, pairwise ~ Algal.treatment | Coral.treatment, adjust = "bonferroni")
# A pulchra - two turfs are sig lower than empty, mimics, and amansia. S punctatus is sig lower than sarg. 
# P rus - two turfs lower than all other treatments, but not from each other. Amansia lower than empty, turf mimc. 

contact_letters_2 <- c('a', 'a', 'a', 'a', 'ab', 'bc', 'c', # A pulchra letters
                    'a', 'ab', 'a', 'b', 'ab', 'c', 'c') # P rus letters





```

## plot figure 2
###VERSION 1 USED IN FINAL PRODUCT

```{r week2 graph, fig.height = 15, fig.width=20}




week2_graph <- ddply(week2, c("Coral.treatment", "Algal.treatment"), numcolwise(mean), na.rm = TRUE )
week2_errors <-  ddply(week2, c("Coral.treatment", "Algal.treatment"), numcolwise(se))
week2_graph$SE <- week2_errors$Yield

# Need to arrange dataframe so connexting letters match correctly
treatment_order <- c('Empty', 'Macroalgae mimic', 'Turf mimic', 'Amansia',
                              'Sargassum', 'S nigricans turf', 'S punctatus turf')

week2_graph <- week2_graph %>% dplyr::arrange(factor(Algal.treatment, levels = treatment_order )) # arrange by algal treatment
week2_graph <- week2_graph %>% dplyr::arrange(factor(Coral.treatment, levels = c('Acropora pulchra', 'Porites rus') )) # arrange by coral
week2_graph$Letters <- contact_letters_2 # add letters
week2_graph$Labels <- c(rep("a. Acropora pulchra\n", 7), rep("b. Porites rus\n", 7))
week2_stats <- data.frame(Coral.treatment = c("Acropora pulchra", "Porites rus"), Text = c("", "Algae: P < E-16\nCoral: P = 0.01\nCoral X Algae: P = 4.6E-04")) # create dataframe so text only appears in bottom facet

## VERSION 1
ggplot(data = week2) +
  geom_boxplot(aes(x = Algal.treatment, y = Yield, colour = Coral.treatment),
               position = position_dodge(0.9), outlier.shape = NA, size = 1.2) +
  scale_x_discrete(limits = treatment_order, 
                       labels = c("Empty",
                              "Turf mimic",
                              "Macroalgal mimic",
                              expression(italic("Amansia")),
                              expression(italic("Sargassum")),
                              expression(italic("S. nigricans")),
                              expression(italic("S. punctatus")))) +
  geom_point(aes(x = Algal.treatment, y = Yield, colour = Coral.treatment),
              size = 3, position = position_jitterdodge(dodge.width = 0.9, jitter.width = 0.3)) +
  scale_colour_manual(values=c("Acropora pulchra" = "dodgerblue",
                               "Porites rus" = "firebrick2"),
                      labels = c(expression(italic("Acropora pulchra")), expression(italic("Porites rus")))
                      ) +
  geom_text(data = week2_graph, aes(x = Algal.treatment, y = 0.7, colour = Coral.treatment, label = Letters),
            position = position_dodge(0.9), size = 12, show.legend = FALSE) +
 geom_text(data = week2_graph, aes(x = Algal.treatment, y = 0.72,
                                    label = c("", "", "", "", "", "**", "***",
                                              "", "", "", "", "", "**", "***")), # add a star for sig interaction 
           size = 12) +
  geom_text(aes(x = 0.7, y = 0.15, label = "Algae: P < E-16\nCoral: P = 0.01\nCoral X Algae: P = 4.6E-04"), hjust = 0, size = 12)+
  labs(x = "", y = "Effective Quantum Yield (Y)\n", color = "Coral species") +
  theme_classic() + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        legend.position = "bottom",
        legend.key.size = unit(2, 'cm'),
        legend.text = element_text(size=25),
        legend.title = element_text(size=28),
        axis.line = element_line(colour = "black"),
        axis.text.x = element_text(color = "grey20", size = 30, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 30, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain")
  )
## TEXT SPACING ON X AXIS EDITED IN PHOTOSHOP

## VERSION 2
ggplot(data = week2) +
  geom_boxplot(aes(x = Algal.treatment, y = Yield, colour = Coral.treatment),
               position = position_dodge(0.9), outlier.shape = NA, size = 1.2) +
  scale_x_discrete(limits = treatment_order, labels = c('Empty', 'Macroalgae\nmimic', 'Turf\nmimic', 'Amansia',
                              'Sargassum', 'S nigricans\nturf', 'S punctatus\nturf')) +
  geom_point(aes(x = Algal.treatment, y = Yield, colour = Coral.treatment),
              size = 3, position = position_jitterdodge(dodge.width = 0.9, jitter.width = 0.3)) +
  scale_colour_manual(values=c("Acropora pulchra" = "dodgerblue",
                               "Porites rus" = "firebrick2") 
                      ) +
  geom_text(data = week2_graph, aes(x = Algal.treatment, y = 0.7, colour = Coral.treatment, label = Letters),
            position = position_dodge(0.9), size = 8) +
 geom_text(data = week2_graph, aes(x = Algal.treatment, y = 0.72,
                                    label = c("", "", "", "", "", "", "",
                                              "", "", "", "", "", "**", "***")), # add a star for sig interaction 
           size = 12) +
  facet_wrap(~Coral.treatment, ncol = 1, 
             labeller = as_labeller(c(`Acropora pulchra` = "a. *A. pulchra*\n",
                                      `Porites rus` = "b. *P. rus*\n"))) +
  geom_text(data = week2_stats, aes(x = 0.7, y = 0.18, label = Text), hjust = 0, size = 9)+
  theme_classic() + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        legend.position = "bottom",
        legend.key.size = unit(2, 'cm'),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        axis.line = element_line(colour = "black"),
        axis.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 28, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 28, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 28, hjust = 0, vjust = 0, face = "plain"),
        strip.background = element_blank(),
        strip.text = element_markdown()
  )

```
# Figure 3 - turf surface chemical extracts
## read and analyze data
```{r surface extracts march}
surface <- read.csv(here::here("data", "turf_extract_pam_data_march.csv")) # read data


# with random effect
surface_beta_mme <- glmmTMB(Yield ~ Algal.treatment*Coral.treatment + (1|Colony.ID), data = surface, beta_family(link = "logit"))


plot(surface_diag <- simulateResiduals(surface_beta_mme)) # non homogeneity of variance in residuals - likely driven by treatments
# Moving on
# EXAMINE FURTHER
plot(residuals(surface_beta_mme) ~ predict(surface_beta_mme)) # some funnelling (reduced variance at higher predicted values)
# NOTE: I think this is fine - it's not really a true continuous distribution since at high values yield sort of caps off
# you would expect reduced variance in "healthy" or unimpacted corals

summary(surface_beta_mme) # low variance between colonies - awesome (std = 0.07)
Anova(surface_beta_mme) # Algal.treatment (X2 = 59.1, P = 1.4e-13; Coral.treatment X2 = 8.9, P = .003; Int X2 = 13.1, P = 0.001)
r.squaredGLMM(surface_beta_mme) # 0.75, 0.82

# within-factor pairwise comparisons - differences between treatments for each coral
emmeans(surface_beta_mme, pairwise ~ Algal.treatment | Coral.treatment, adjust = "bonferroni", type = "response")
# Acropora: both turfs sig lower than control
# P rus: only S punctatus sig lower than control

# within-factor pairwise comparisons - differences between corals in each treatment
emmeans(surface_beta_mme, pairwise ~ Coral.treatment | Algal.treatment, adjust = "bonferroni")
# Sig difference between corals for both turf trt

surface_letters = c('a', 'b', 'b', # A pulchra letters from emmeans
                    'a', 'ab', 'b') # P rus letters from emmeans

```

## plot figure 3

```{r graph surface extracts, fig.height = 16, fig.width = 20}
surface_means <- ddply(surface, c("Coral.treatment", "Algal.treatment"), numcolwise(mean), na.rm = TRUE )
errors <-  ddply(surface, c("Coral.treatment", "Algal.treatment"), numcolwise(se))
surface_means$SE <- errors$Yield
surface_means$Letters <- surface_letters
surface_means$Coral_contrast <- c("", "**", "**",
                                 "", "**", "**") # for coral comparisons


ggplot(data = surface) +
  geom_boxplot(aes(x = Algal.treatment, y = Yield, colour = Coral.treatment),
               position = position_dodge(0.9), outlier.shape = NA, size = 1.1) +
    geom_point(data = surface, aes(x = Algal.treatment, y = Yield, colour = Coral.treatment),
              size = 3, position = position_jitterdodge(dodge.width = 0.9, jitter.width = 0.3)) +
  geom_text(data = surface_means, aes(x = Algal.treatment, y = 0.7, colour = Coral.treatment, label = Letters), position = position_dodge(0.9), size = 12, show.legend = FALSE) +
  geom_text(aes(x = 0.5, y = 0.21, label = "Algae: P < 1.4E-13\nCoral: P = 0.003\nCoral X Algae: P = 0.001"), hjust = 0, size = 14)+
  geom_text(data = surface_means, aes(x = Algal.treatment, y = 0.72, label = Coral_contrast), size = 12) +
  scale_colour_manual(values=c("Acropora pulchra" = "dodgerblue",
                               "Porites rus" = "firebrick2"),
                      labels = c(expression(italic("A. pulchra")), expression(italic("P. rus")))
                      ) +
   scale_x_discrete(labels = c("\nControl", expression(italic("\nS. nigricans")), expression(italic("\nS. punctatus")) ) ) +
  labs(x = "", y = "Effective Quantum Yield (Y)\n", color = "Coral species") +
  theme_classic() + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        legend.position = "bottom",
        legend.key.size = unit(2, 'cm'),
        legend.text = element_text(size=38),
        legend.title = element_text(size=38),
        axis.line = element_line(colour = "black"),
        axis.text.x = element_text(color = "grey20", size = 38, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 30, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 38, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain")
  )



```

# Figure 4 - seasonal extracts experiment 
## read data and model
```{r winter surface, fig.height=10, fig.width=10}
winter <- read.csv(here::here("data", "turf_extract_pam_data_winter.csv"), stringsAsFactors = T)
# levels(winter$Algal.treatment)[levels(winter$Algal.treatment)=="Summer S nigricans turf"] <- "March S nigricans turf"

# basic viz
bwplot(Yield ~ Algal.treatment | Coral.treatment, data = winter,
       par.settings = list(box.dot = list(col = "black", cex=2),
                           box.umbrella = list(col = "black"),
                           box.rectangle = list(col = "black"),
                           plot.symbol   = list(cex = 2, col = 1)))


# model
beta_mme <- glmmTMB(Yield ~ Algal.treatment*Coral.treatment + (1|Colony.ID), data = winter, beta_family(link = "logit"))
plot(winter_diag <- simulateResiduals(beta_mme)) # no problems detected in model
Anova(beta_mme) # Difference in yield between corals (X2 = 8.1, P < 0.004), but not between algal treatments (X2 = 6.1, P = 0.2) or interaction (X2 = 5.6, P = 0.2)
r.squaredGLMM(beta_mme) # 0.36, 0.57
summary(beta_mme) # some variance at colony level (std = 0.09)


emmeans(beta_mme, pairwise ~ Algal.treatment | Coral.treatment, type = "response") # no sig differences
letters_winter = c('a', 'a', 'a', 'a', 'a',
                         'a', 'a', 'a', 'a', 'a')




```

## plot figure 4
```{r figure 4 plot, fig.height = 16, fig.width = 20}



## PLOT

ggplot(data = winter) +
  geom_boxplot(aes(x = Algal.treatment, y = Yield, colour = Coral.treatment),
               position = position_dodge(0.9), outlier.shape = NA, size = 1.1) +
  geom_point(data = winter, aes(x = Algal.treatment, y = Yield, colour = Coral.treatment),
              size = 3, position = position_jitterdodge(dodge.width = 0.9, jitter.width = 0.3)) +
  geom_text(aes(x = 0.5, y = 0.23, label = "Algae: P = 0.2\nCoral: P = 0.004\nCoral X Algae: P = 0.2"), hjust = 0, size = 14)+
  scale_y_continuous(limits = c(0.2,0.71)) +
  scale_x_discrete(labels = c("Control", 
                              "March\nS. nigricans\nturf extract",
                              "March\nS. punctatus\nturf extract",
                              "September\nS. nigricans\nturf extract",
                              "September\nS. punctatus\nturf extract")) +
  scale_colour_manual(values=c("Acropora pulchra" = "dodgerblue",
                               "Porites rus" = "firebrick2"),
                      label = c(expression(italic("Acropora pulchra")), expression(italic("Porites rus")))
                      ) +
  labs(x = "", y = "Effective Quantum Yield (Y)\n", color = "Coral species") +
  theme_classic() + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        legend.position = "bottom",
        legend.key.size = unit(2, 'cm'),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        axis.line = element_line(colour = "black"),
        axis.text.x = element_text(color = "grey20", size = 28, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 28, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 28, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 28, hjust = .5, vjust = 0, face = "plain")
  )
# AXIS TEXT CORRECTED FOR ITALICS IN PHOTOSHOP

```
# Microbial analyses start
```{r read and clean microbial data}
# read in acropora SVs
acropora_svs <- read.csv(here::here("data", "acropora_rarefied_table.csv"), row.names = 1)
acropora_svs_t <- as.data.frame(t(acropora_svs)) # transpose sv dataframe
str(acropora_svs_t) # make sure it transpose properly
acropora_svs_t <- acropora_svs_t[order(row.names(acropora_svs_t)), ] # reorder rows to fit meta
acropora_svsP <- otu_table(acropora_svs, taxa_are_rows = TRUE)  # coerce SV csv into otu table object for phyloseq

porites_svs <- read.csv(here::here("data", "porites_rarefied_table.csv"), row.names = 1)

porites_svs_t <- as.data.frame(t(porites_svs)) 

str(porites_svs_t) # make sure it transpose properly
porites_svs_t <- porites_svs_t[order(row.names(porites_svs_t)), ] # reorder rows to fit meta
porites_svsP <- otu_table(porites_svs, taxa_are_rows = TRUE)

meta = read.csv(here::here("data", "coral_algal_contact_microbial_metadata_no_blanks.csv") , row.names=1)# sample ID as rowname
meta = meta[order(row.names(meta)), ]

acropora_meta <- subset(meta, Coral.trt == "Acropora") # subset for creating phyloseq objects
acropora_metaP <- sample_data(acropora_meta) # specify metadata as sampledata for phyloseq

porites_meta <- subset(meta, Coral.trt == "Porites") # subset for creating phyloseq objects

porites_metaP <- sample_data(porites_meta) # specify metadata as sampledata for phyloseq


taxa <- read.csv(here::here("data", "microbial_taxonomy.csv"), row.names=1) # read in taxonomy data
taxa <- as.matrix(taxa) # convert to matrix
taxaP <- tax_table(taxa) # convert taxonomy into taxonomyTable object for phyloseq

tree <- read_qza(here::here("data", "rooted_tree.qza")) # read in tree






acropora <- phyloseq(acropora_svsP, taxaP, acropora_metaP, tree$data) # compile the three objects above into data for phyloseq
porites <- phyloseq(porites_svsP, taxaP, porites_metaP, tree$data) # repeat for porites

#Sort samples on total read count, remove <5k reads, remove any OTUs seen in only those samples
sort(phyloseq::sample_sums(acropora)) # check if there are samples with <5k reads 
# all of these have the same number of samples (14580) because of qiime cleaning


#Convert to relative abundance
acropora_rel_abund <- phyloseq::transform_sample_counts(acropora, function(x){x / sum(x)})

porites_rel_abund <- phyloseq::transform_sample_counts(porites, function(x){x / sum(x)})


# Output OTU tables
acropora_otu <- data.frame(phyloseq::otu_table(acropora))
porites_otu <- data.frame(phyloseq::otu_table(porites))




```




## dendrograms - not used in final MS
```{r acropora only dendrogram}

#Extract OTU table and compute BC
acropora_rel_otu <- data.frame(phyloseq::otu_table(acropora_rel_abund))
acropora_rel_otu <- t(acropora_rel_otu)
acropora_bc_dist <- vegan::vegdist(acropora_rel_otu, method = "bray")

#Save as dendrogram
ward <- as.dendrogram(hclust(acropora_bc_dist, method = "ward.D2"))
#Provide color codes
meta <- data.frame(phyloseq::sample_data(acropora_rel_abund))
colorCode <- c(Empty = "grey",
               `Turf mimic` = "black",
               `Macroalgae mimic` = "green",
               Amansia = "red",
               Sargassum = "brown",
               `S nigricans` = "blue",
               `S punctatus` = "navy")

labels_colors(ward) <- colorCode[meta$Algal.trt][order.dendrogram(ward)]
#Plot
plot(ward)
# No real clustering

```


```{r porites only dendrogram}

#Extract OTU table and compute BC
porites_rel_otu <- data.frame(phyloseq::otu_table(porites_rel_abund))
porites_rel_otu <- t(porites_rel_otu)
porites_bc_dist <- vegan::vegdist(porites_rel_otu, method = "bray")

#Save as dendrogram
ward <- as.dendrogram(hclust(porites_bc_dist, method = "ward.D2"))
#Provide color codes
meta <- data.frame(phyloseq::sample_data(porites_rel_abund))
colorCode <- c(Empty = "grey",
               `Turf mimic` = "black",
               `Macroalgae mimic` = "green",
               Amansia = "red",
               Sargassum = "brown",
               `S nigricans` = "blue",
               `S punctatus` = "navy")

labels_colors(ward) <- colorCode[meta$Algal.trt][order.dendrogram(ward)]
#Plot
plot(ward)
# Some clustering among turf trts

```

## Figure 5 - alpha diversity models
### Acropora
```{r acropora alpha diversity}



#Generate a data.frame with alpha div measures
acropora_adiv <- data.frame(
  "Observed" = phyloseq::estimate_richness(acropora, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(acropora, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(acropora, measures = "Simpson"),
  "Algal.trt" = phyloseq::sample_data(acropora)$Algal.trt, 
  "Coral.colony.ID" = phyloseq::sample_data(acropora)$Coral.colony.ID)
head(acropora_adiv)


# check dist
hist(acropora_adiv$Shannon) # highly skewed
hist(acropora_adiv$Observed)

plot_richness(acropora, measures = c("Observed", "Shannon", "Simpson"))

#Plot acropora_adiv measures
acropora_adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon"))) %>%
  ggplot(aes(x = Algal.trt, y = value), size = 2) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Algal.trt), height = 0, width = .2, size = 2) +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none",
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
# large spread in both richness and diversity between treatments


# check for Coral.colony.IDs
acropora_adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon"))) %>%
  ggplot(aes(x = Coral.colony.ID, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Algal.trt), height = 0, width = .2) +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none",
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# Acropora 1 and 2 stand out


#Summarize
acropora_adiv %>%
  group_by(Algal.trt) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon))
# medians look higher in turf trt


acropora_adiv %>%
  group_by(Coral.colony.ID) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon))

acropora_adiv %>%
  group_by(Algal.trt) %>%
  dplyr::summarise(mean_observed = mean(Observed),
            mean_shannon = mean(Shannon))


# mixed gamma reg
acropora_gamma <- glmmTMB(Shannon ~ Algal.trt  + (1| Coral.colony.ID) , data = acropora_adiv,
                       family = Gamma(link = "log") )
summary(acropora_gamma)
Anova(acropora_gamma) # X2 = 35.6 P = 3.339e-06
plot(residuals(acropora_gamma) ~ predict(acropora_gamma)) # looks good
hist(residuals(acropora_gamma))
# some skew in residuals driven by skew in data
# this seems likely to be real and natural variation in the data (treatment effects)
# otherwise model diags look great - continuing here
plot(allEffects(acropora_gamma))
acropora_gamma_comp <- emmeans(acropora_gamma, pairwise ~ Algal.trt)
acropora_gamma_letters <- cld(object = acropora_gamma_comp,
                      adjust = "Tukey",
                      Letters = letters,
                      alpha = 0.05)

# poisson richness
acropora_poisson <- glmmTMB(Observed ~ Algal.trt  + (1| Coral.colony.ID) , data = acropora_adiv,
                       family = poisson )
summary(acropora_poisson)
plot(residuals(acropora_poisson) ~ predict(acropora_poisson)) # highly heteroskedastic
hist(residuals(acropora_poisson)) 
acropora_poisson_comp <- emmeans(acropora_poisson, pairwise ~ Algal.trt)
acropora_poisson_letters <- cld(object = acropora_poisson_comp,
                      adjust = "Tukey",
                      Letters = letters,
                      alpha = 0.05)

# nb richness
acropora_nb <- glmmTMB(Observed ~ Algal.trt  + (1| Coral.colony.ID) , family = nbinom2, data = acropora_adiv ) 
summary(acropora_nb)
Anova(acropora_nb) # X2 = 41.407  df = 6  P = 2.408e-07
plot(residuals(acropora_nb) ~ predict(acropora_nb)) 
hist(residuals(acropora_nb)) # still some issues but on the whole much better and addresses overdispersion brought on from skew
acropora_nb_comp <- emmeans(acropora_poisson, pairwise ~ Algal.trt)
acropora_nb_letters <- cld(object = acropora_nb_comp,
                      adjust = "Tukey",
                      Letters = letters,
                      alpha = 0.05)
```


### Plot - figure 5ce alpha diversity  acropora
```{r acropora shannon and observed plot, fig.height = 15, fig.width=18}
(acropora_observed_ggplot <- 
  acropora_adiv %>%
  gather(key = metric, value = value, c("Observed")) %>%
  mutate(metric = factor(metric, levels = c("Observed"))) %>%
  ggplot(aes(x = Algal.trt, y = value)) +
  geom_boxplot(outlier.color = NA, size = 1.2) +
  geom_jitter(aes(color = Algal.trt), height = 0, width = .2, size = 4, colour = "black") +
  geom_text(data = acropora_nb_letters, aes(x = Algal.trt, y = 1050, label = .group), size = 8) +
  scale_x_discrete(limits = c("Empty", "Turf mimic", "Macroalgae mimic", "Amansia", "Sargassum", "S nigricans", "S punctatus"),
    labels = c("Empty" = "Zip-tie",
                              "Turf mimic" = "Turf mimic",
                              "Macroalgae mimic" = "Macroalgal mimic",
                              "Amansia" = expression(italic("Amansia")),
                              "Sargassum" = expression(italic("Sargassum")),
                              "S nigricans" = expression(italic("S. nigricans")),
                              "S punctatus" = expression(italic("S. punctatus")))
                   ) +
  labs(x = "", y = "Observed SVs", title = "c. *A. pulchra* Richness\n") +
  theme(legend.key.size = unit(6, 'cm'),
        legend.key=element_rect(fill="white"),
        legend.position = 'bottom',
        legend.text = element_text(size=30),
        legend.title = element_text(size=30),
        legend.box = 'horizontal', # make legend horizontal
        legend.box.background = element_rect(colour = "black"), # add box around legend
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_markdown(color = "grey20", size = 32, vjust = 0, face = "plain"),
        axis.line.y = element_line(colour = "black"),
        axis.text.y = element_text(color = "grey20", size = 28, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.line.x = element_line(colour = "black"),
        axis.title.x = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.text.x = element_text(color = "grey20", size = 28, angle = 70, vjust = 1, hjust=1, face = "plain"))) + 
  guides(colour=guide_legend(title.position="top", # make color legend title go above legend
                                     title.hjust =0.5, nrow = 2, override.aes = list(size=10)) )


(acropora_adiv_ggplot = 
  acropora_adiv %>%
  gather(key = metric, value = value, c("Shannon")) %>%
  mutate(metric = factor(metric, levels = c("Shannon"))) %>%
  ggplot(aes(x = Algal.trt, y = value)) +
  geom_boxplot(outlier.color = NA, size = 1.2) +
  geom_jitter(aes(color = Algal.trt), height = 0, width = .2, size = 4, colour = "black") +
  geom_text(data = acropora_gamma_letters, aes(x = Algal.trt, y = 6.5, label = .group), size = 8) +
  scale_x_discrete(limits = c("Empty", "Turf mimic", "Macroalgae mimic", "Amansia", "Sargassum", "S nigricans", "S punctatus"),
    labels = c("Empty" = "Zip-tie",
                              "Turf mimic" = "Turf mimic",
                              "Macroalgae mimic" = "Macroalgal mimic",
                              "Amansia" = expression(italic("Amansia")),
                              "Sargassum" = expression(italic("Sargassum")),
                              "S nigricans" = expression(italic("S. nigricans")),
                              "S punctatus" = expression(italic("S. punctatus")))
                   ) +
  labs(x = "", y = "Shannon diversity\n", title = "e. *A. pulchra* Shannon diversity\n") +
  theme(legend.position="none",
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_markdown(color = "grey20", size = 32, vjust = 0, face = "plain"),
        axis.line.y = element_line(colour = "black"),
        axis.text.y = element_text(color = "grey20", size = 28, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.line.x = element_line(colour = "black"),
        axis.title.x = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.text.x = element_text(color = "grey20", size = 28, angle = 70, vjust = 1, hjust=1, face = "plain")))
```

### Porites

```{r porites alpha diversity}


#Generate a data.frame with alpha div measures
porites_adiv <- data.frame(
  "Observed" = phyloseq::estimate_richness(porites, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(porites, measures = "Shannon"),
  "Algal.trt" = phyloseq::sample_data(porites)$Algal.trt, 
  "Coral.colony.ID" = phyloseq::sample_data(porites)$Coral.colony.ID)
head(porites_adiv)


# vizualize
hist(porites_adiv$Observed) # not as skewed as acropora but still not great. more uniform

plot_richness(porites, measures = c("Observed", "Shannon", "InvSimpson"))

#Plot porites_adiv measures
porites_adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon"))) %>%
  ggplot(aes(x = Algal.trt, y = value), size = 2) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Algal.trt), height = 0, width = .2, size = 2) +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none",
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
# results completely different this time


# check for Coral.colony.IDs
porites_adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon"))) %>%
  ggplot(aes(x = Coral.colony.ID, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Algal.trt), height = 0, width = .2) +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none",
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# porites 11 looks off 


#Summarize
porites_adiv %>%
  group_by(Algal.trt) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon))
# empty WAY higher this time around... what changed?


porites_adiv %>%
  group_by(Coral.colony.ID) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon))
# porites 11 weird

porites_adiv %>%
  group_by(Algal.trt) %>%
  dplyr::summarise(mean_observed = mean(Observed),
            mean_shannon = mean(Shannon))




# mixed gamma reg
porites_gamma <- glmmTMB(Shannon ~ Algal.trt  + (1| Coral.colony.ID) , data = porites_adiv,
                       family = Gamma(link = "log") )
summary(porites_gamma)
Anova(porites_gamma) # X2 = 14.3  P = 0.03
plot(residuals(porites_gamma) ~ predict(porites_gamma))
hist(residuals(porites_gamma)) # looks ok - maybe a little lumpy but not bad
qqnorm(residuals(porites_gamma))
qqline(residuals(porites_gamma))
porites_gamma_comp <- emmeans(porites_gamma, pairwise ~ Algal.trt)
porites_gamma_letters <- cld(object = porites_gamma_comp,
                      adjust = "Tukey",
                      Letters = letters,
                      alpha = 0.05)

# porites poisson
porites_poisson <- glmmTMB(Observed ~ Algal.trt  + (1| Coral.colony.ID) , data = porites_adiv,
                       family = poisson )
summary(porites_poisson)
Anova(porites_poisson) # X2 = 796.2  P < 2.2e-16
plot(residuals(porites_poisson) ~ predict(porites_poisson))
hist(residuals(porites_poisson)) # looks ok
porites_poisson_comp = emmeans(porites_poisson, pairwise ~ Algal.trt)
porites_poisson_letters <- cld(object = porites_poisson_comp,
                      adjust = "Tukey",
                      Letters = letters,
                      alpha = 0.05)




```

### Plot figure 5df
```{r porites shannon and richness plot, fig.height = 15, fig.width=18}
(porites_observed_ggplot <- 
  porites_adiv %>%
  gather(key = metric, value = value, c("Observed")) %>%
  mutate(metric = factor(metric, levels = c("Observed"))) %>%
  ggplot(aes(x = Algal.trt, y = value)) +
  geom_boxplot(outlier.color = NA, size = 1.2) +
  geom_jitter(aes(color = Algal.trt), height = 0, width = .2, size = 4, colour = "black") +
  geom_text(data = porites_poisson_letters, aes(x = Algal.trt, y = 1100, label = .group), size = 8) +
  scale_x_discrete(limits = c("Empty", "Turf mimic", "Macroalgae mimic", "Amansia", "Sargassum", "S nigricans", "S punctatus"),
    labels = c("Empty" = "Zip-tie",
                              "Turf mimic" = "Turf mimic",
                              "Macroalgae mimic" = "Macroalgal mimic",
                              "Amansia" = expression(italic("Amansia")),
                              "Sargassum" = expression(italic("Sargassum")),
                              "S nigricans" = expression(italic("S. nigricans")),
                              "S punctatus" = expression(italic("S. punctatus")))
                   ) +
  labs(x = "", y = "", title = "d. *P. rus* Richness\n") +
  theme(legend.position="none",
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_markdown(color = "grey20", size = 32, vjust = 0, face = "plain"),
        axis.line.y = element_line(colour = "black"),
        axis.text.y = element_text(color = "grey20", size = 28, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.line.x = element_line(colour = "black"),
        axis.title.x = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.text.x = element_text(color = "grey20", size = 28, angle = 70, vjust = 1, hjust=1, face = "plain")))


(porites_adiv_ggplot = 
  porites_adiv %>%
  gather(key = metric, value = value, c("Shannon")) %>%
  mutate(metric = factor(metric, levels = c("Shannon"))) %>%
  ggplot(aes(x = Algal.trt, y = value)) +
  geom_boxplot(outlier.color = NA, size = 1.2) +
  geom_jitter(aes(color = Algal.trt), height = 0, width = .2, size = 4, colour = "black") +
    geom_text(data = porites_gamma_letters, aes(x = Algal.trt, y = 6.5, label = .group), size = 8) +
  scale_x_discrete(limits = c("Empty", "Turf mimic", "Macroalgae mimic", "Amansia", "Sargassum", "S nigricans", "S punctatus"),
    labels = c("Empty" = "Zip-tie",
                              "Turf mimic" = "Turf mimic",
                              "Macroalgae mimic" = "Macroalgal mimic",
                              "Amansia" = expression(italic("Amansia")),
                              "Sargassum" = expression(italic("Sargassum")),
                              "S nigricans" = expression(italic("S. nigricans")),
                              "S punctatus" = expression(italic("S. punctatus")))
                   ) +
  labs(x = "", y = "\n", title = "f. *P. rus* Shannon diversity\n") +
  theme(legend.position="none",
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_markdown(color = "grey20", size = 32, vjust = 0, face = "plain"),
        axis.line.y = element_line(colour = "black"),
        axis.text.y = element_text(color = "grey20", size = 28, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.line.x = element_line(colour = "black"),
        axis.title.x = element_text(color = "grey20", size = 32, hjust = .5, vjust = 0, face = "plain"),
        axis.text.x = element_text(color = "grey20", size = 28, angle = 70, vjust = 1, hjust=1, face = "plain")))

```

```{r alpha diversity panel graph, fig.height = 18, fig.width=18}


ggarrange(acropora_observed_ggplot + theme(axis.text.x=element_blank()), acropora_adiv_ggplot + theme(axis.text.x=element_blank()),
          porites_observed_ggplot, porites_adiv_ggplot, ncol=2, nrow=2, heights = c(1,1.5), widths = c(1,1),
          common.legend = TRUE, legend="bottom") 
# creates lots of empty space
# crop in photoshop

```


### Figure S3 - binary jaccard


```{r acropora jaccard, fig.height=10, fig.width=10}
acropora_binary <- microbiome::transform(acropora, "pa")  # convert acropora to binary presence absence following qiime method
acropora_jaccard_matrix <- phyloseq::distance(acropora_binary, method = "jaccard") 


anosim(acropora_jaccard_matrix, phyloseq::sample_data(acropora_binary)$Algal.trt) # R = 0.23, P = 0.001
adonis2(acropora_jaccard_matrix ~ phyloseq::sample_data(acropora_binary)$Algal.trt) # R2 = 0.1, F = 1.4, P < 0.001


acropora_jaccard <- ordinate(acropora_binary, distance = "jaccard", method = "PCoA")
acropora_jaccard1 <- acropora_jaccard$values$Eigenvalues[1] / sum(acropora_jaccard$values$Eigenvalues)
acropora_jaccard2 <- acropora_jaccard$values$Eigenvalues[2]  / sum(acropora_jaccard$values$Eigenvalues)



# Check to see if ordination matches qiime emperor output
(acropora_jaccard_plot <- plot_ordination(acropora_binary, acropora_jaccard, type = "samples", color = "Algal.trt")  + 
  theme_bw() +
  labs(title = "a. *A. pulchra* Jaccard\n") +
  geom_text(aes(x = -0.55, y = -0.36, label = "PERMANOVA\nR = 0.1, P < 0.001"), size = 6, hjust = 0, colour = 'black') +
  geom_point(size = 4) +
#  coord_fixed() +
  scale_color_manual(values = c(Empty = "grey",
               `Turf mimic` = "black",
               `Macroalgae mimic` = "forestgreen",
               Amansia = "red",
               Sargassum = "brown",
               `S nigricans` = "blue",
               `S punctatus` = "skyblue"),
               breaks = c("Empty", "Turf mimic", "Macroalgae mimic", "Amansia", "Sargassum", "S nigricans", "S punctatus"),
               labels = c("Zip-tie", "Turf mimic", "Macroalgal mimic", expression(italic("Amansia")), expression(italic("Sargassum")),
                          expression(italic("S. nigricans")), expression(italic("S. punctatus" )) ) 
               ) +
  theme(panel.background = element_blank(), 
        panel.grid.major = element_blank(),  #remove major-grid labels
        panel.grid.minor = element_blank(),  #remove minor-grid labels
        plot.background = element_blank(),
        plot.title = element_markdown(color = "grey20", size = 25, hjust = 0, vjust = 0, face = "plain"),
        axis.text.x = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain")) +
    stat_ellipse(aes(group = Algal.trt), linetype = 1, type = "norm")
) 

# it does


# PAIRWISE COMPARISONS FOR ANOSIM (https://www.yanh.org/2021/01/01/microbiome-r/)

cbn <- combn(x=unique(phyloseq::sample_data(acropora)$Algal.trt), m = 2) # Create dataframe specifying possible comparisons


acropora.jaccard.p <- c() # Create vector for outputting P values

for(i in 1:ncol(cbn)){ # for each possible comparison
  ps.subs <- subset_samples(acropora, Algal.trt %in% cbn[,i]) # subset the data in the trt for that comparison
  ps.subs.binary <- microbiome::transform(ps.subs, "pa") 
  metadata_sub <- data.frame(sample_data(ps.subs)) # specify the metadata for that subset
  
  permanova_pairwise <- anosim(phyloseq::distance(ps.subs.binary, method = "jaccard"),
                               metadata_sub$Algal.trt) # perform anosim for the data based on those two treatments
  acropora.jaccard.p <- c(acropora.jaccard.p, permanova_pairwise$signif[1]) # append the p value for that comparison to the created p value dataframe
}

acropora.jaccard.p.adj <- p.adjust(acropora.jaccard.p, method = "bonferroni") # adjust the p value for multiple comparisons
acropora.jaccard.p.table <- as.data.frame(cbind.data.frame(t(cbn), p=acropora.jaccard.p, p.adj=acropora.jaccard.p.adj)) # create data table containing p value comparisons 
# SN and SP different from empty, TM, and MM. SP also different from amansia
# write.csv(acropora.jaccard.p.table, "/Users/noamaltman-kurosaki/iCloud Drive (Archive) - 1/Documents/Grad school (general)/PhD General/Coral-algal contact experiment/acropora pairwise anosim jaccard sqrt transformed.csv")


acropora.jaccard.adonisp <- c() # Create vector for outputting P values

for(i in 1:ncol(cbn)){ # for each possible comparison
  ps.subs <- subset_samples(acropora, Algal.trt %in% cbn[,i]) # subset the data in the trt for that comparison
  ps.subs.binary <- microbiome::transform(ps.subs, "pa") 
  metadata_sub <- data.frame(sample_data(ps.subs)) # specify the metadata for that subset
  
  permanova_pairwise <- adonis2(phyloseq::distance(ps.subs.binary, method = "jaccard") ~
                               metadata_sub$Algal.trt) # perform anosim for the data based on those two treatments
  acropora.jaccard.adonisp <- c(acropora.jaccard.adonisp,  permanova_pairwise$`Pr(>F)`[1]) # append the p value for that comparison to the created p value dataframe
}

acropora.jaccard.adonisp.adj <- p.adjust(acropora.jaccard.adonisp, method = "bonferroni") # adjust the p value for multiple comparisons
acropora.jaccard.adonisp.table <- as.data.frame(cbind.data.frame(t(cbn), p=acropora.jaccard.adonisp, p.adj=acropora.jaccard.adonisp.adj)) # create data table containing p value comparisons 

# write.csv(acropora.jaccard.adonisp.table, "/Users/noamaltman-kurosaki/iCloud Drive (Archive) - 1/Documents/Grad school (general)/PhD General/Coral-algal contact experiment/acropora pairwise anosim jaccard sqrt transformed.csv")

#Dispersion test
acropora_dispr <- vegan::betadisper(acropora_jaccard_matrix, phyloseq::sample_data(acropora)$Algal.trt)
acropora_dispr # 

# plot of dispersion test
plot(acropora_dispr, main = "Ordination Centroids and Dispersion Labeled: Aitchison Distance", sub = "")
boxplot(acropora_dispr, main = "", xlab = "") # turf trts generally furthest from centroid

vegan::permutest(acropora_dispr) # F = 0.6, P = 0.7



## Tukey's Honest Significant Differences
(mod.HSD <- TukeyHSD(acropora_dispr)) # no sig differences

```

```{r porites jaccard, fig.height=10, fig.width=10}
porites_binary <- microbiome::transform(porites, "pa")  # convert porites to binary presence absence following qiime method
porites_jaccard_matrix <- phyloseq::distance(porites_binary, method = "jaccard") 


anosim(porites_jaccard_matrix, phyloseq::sample_data(porites_binary)$Algal.trt) # R = 0.24, P = 0.001
adonis2(porites_jaccard_matrix ~ phyloseq::sample_data(porites_binary)$Algal.trt) # R2 = 0.11, F = 1.5, P < 0.001


porites_jaccard <- ordinate(porites_binary, distance = "jaccard", method = "PCoA")
porites_jaccard1 <- porites_jaccard$values$Eigenvalues[1] / sum(porites_jaccard$values$Eigenvalues)
porites_jaccard2 <- porites_jaccard$values$Eigenvalues[2]  / sum(porites_jaccard$values$Eigenvalues)

# Check to see if ordination matches qiime emperor output
(porites_jaccard_plot <- plot_ordination(porites_binary, porites_jaccard, type = "samples", color = "Algal.trt")  + 
    geom_point(size = 4) +
    geom_text(aes(x = 0.8, y = 0.45, label = "PERMANOVA\nR = 0.11, P < 0.001"), colour = "black", hjust = 1, size = 6) +
  theme_bw() +
  labs(title = "b. *P. rus* Jaccard\n") +
  geom_point(size = 2) +
#  coord_fixed() +
  scale_color_manual(values = c(Empty = "grey",
               `Turf mimic` = "black",
               `Macroalgae mimic` = "forestgreen",
               Amansia = "red",
               Sargassum = "brown",
               `S nigricans` = "blue",
               `S punctatus` = "skyblue"),
               breaks = c("Empty", "Turf mimic", "Macroalgae mimic", "Amansia", "Sargassum", "S nigricans", "S punctatus"),
               labels = c("Zip-tie", "Turf mimic", "Macroalgal mimic", expression(italic("Amansia")), expression(italic("Sargassum")),
                          expression(italic("S. nigricans")), expression(italic("S. punctatus" )) ) 
               ) +
  theme(panel.background = element_blank(), 
        panel.grid.major = element_blank(),  #remove major-grid labels
        panel.grid.minor = element_blank(),  #remove minor-grid labels
        plot.background = element_blank(),
                plot.title = element_markdown(color = "grey20", size = 25, hjust = 0, vjust = 0, face = "plain"),
        axis.text.x = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        legend.position = 'bottom')  +
    stat_ellipse(aes(group = Algal.trt), linetype = 1, type = "norm")
)

# We're good!

# PAIRWISE COMPARISONS FOR ANOSIM (https://www.yanh.org/2021/01/01/microbiome-r/)

cbn <- combn(x=unique(phyloseq::sample_data(porites)$Algal.trt), m = 2) # Create dataframe specifying possible comparisons


porites.jaccard.p <- c() # Create vector for outputting P values

for(i in 1:ncol(cbn)){ # for each possible comparison
  ps.subs <- subset_samples(porites, Algal.trt %in% cbn[,i]) # subset the data in the trt for that comparison
  ps.subs.binary <- microbiome::transform(ps.subs, "pa") 
  metadata_sub <- data.frame(sample_data(ps.subs)) # specify the metadata for that subset
  
  permanova_pairwise <- anosim(phyloseq::distance(ps.subs.binary, method = "jaccard"),
                               metadata_sub$Algal.trt) # perform anosim for the data based on those two treatments
  porites.jaccard.p <- c(porites.jaccard.p, permanova_pairwise$signif[1]) # append the p value for that comparison to the created p value dataframe
}

porites.jaccard.p.adj <- p.adjust(porites.jaccard.p, method = "bonferroni") # adjust the p value for multiple comparisons
porites.jaccard.p.table <- as.data.frame(cbind.data.frame(t(cbn), p=porites.jaccard.p, p.adj=porites.jaccard.p.adj)) # create data table containing p value comparisons 
# SN and SP different from empty and TM. Amansia also different from empty. SN and amansia different. 
# write.csv(porites.jaccard.p.table, "/Users/noamaltman-kurosaki/iCloud Drive (Archive) - 1/Documents/Grad school (general)/PhD General/Coral-algal contact experiment/porites pairwise anosim jaccard sqrt transformed.csv")



porites.jaccard.adonisp <- c() # Create vector for outputting P values

for(i in 1:ncol(cbn)){ # for each possible comparison
  ps.subs <- subset_samples(porites, Algal.trt %in% cbn[,i]) # subset the data in the trt for that comparison
  ps.subs.binary <- microbiome::transform(ps.subs, "pa") 
  metadata_sub <- data.frame(sample_data(ps.subs)) # specify the metadata for that subset
  
  permanova_pairwise <- adonis2(phyloseq::distance(ps.subs.binary, method = "jaccard") ~
                               metadata_sub$Algal.trt) # perform anosim for the data based on those two treatments
  porites.jaccard.adonisp <- c(porites.jaccard.adonisp,  permanova_pairwise$`Pr(>F)`[1]) # append the p value for that comparison to the created p value dataframe
}

porites.jaccard.adonisp.adj <- p.adjust(porites.jaccard.adonisp, method = "bonferroni") # adjust the p value for multiple comparisons
porites.jaccard.adonisp.table <- as.data.frame(cbind.data.frame(t(cbn), p=porites.jaccard.adonisp, p.adj=porites.jaccard.adonisp.adj)) # create data table containing p value comparisons 

# write.csv(porites.jaccard.adonisp.table, "/Users/noamaltman-kurosaki/iCloud Drive (Archive) - 1/Documents/Grad school (general)/PhD General/Coral-algal contact experiment/porites pairwise anosim jaccard sqrt transformed.csv")

#Dispersion test
porites_dispr <- vegan::betadisper(porites_jaccard_matrix, phyloseq::sample_data(porites)$Algal.trt)
porites_dispr # 

# plot of dispersion test
plot(porites_dispr, main = "Ordination Centroids and Dispersion Labeled: Aitchison Distance", sub = "")
boxplot(porites_dispr, main = "", xlab = "") # turf trts generally furthest from centroid
vegan::permutest(porites_dispr) # F = 0.7, P = 0.7




## Tukey's Honest Significant Differences
(mod.HSD <- TukeyHSD(porites_dispr)) # no sig differences

```
### Figure 5ab - bray curtis ordination

```{r acropora bray curtis, fig.height=10, fig.width=10}

# qiime does not transform data prior to calculating bray-curtis dissimlarity matrix
# https://forum.qiime2.org/t/transformations-prior-to-diversity-plug-in-distance-calculations/17709/11
# hellinger/sqrt transformation on relative abundance data appropriate with large number of taxa and high variance in relative abundances
# https://stats.stackexchange.com/questions/394572/what-is-the-most-appropriate-transformation-method-for-performing-analyses-on-sp
acropora_sqrt <- microbiome::transform(acropora_rel_abund, "hellinger") 

acropora_bray_matrix <- phyloseq::distance(acropora_sqrt, method = "bray") 

anosim(acropora_bray_matrix, phyloseq::sample_data(acropora_sqrt)$Algal.trt) # R = 0.11, P = 0.001
adonis2(acropora_bray_matrix ~ phyloseq::sample_data(acropora_sqrt)$Algal.trt) # R2 = 0.13, F = 2.1, P = 0.001

# PAIRWISE COMPARISONS FOR ANOSIM (https://www.yanh.org/2021/01/01/microbiome-r/)

cbn <- combn(x=unique(phyloseq::sample_data(acropora)$Algal.trt), m = 2) # Create dataframe specifying possible comparisons


acropora.bray.p <- c() # Create vector for outputting P values

for(i in 1:ncol(cbn)){ # for each possible comparison
  ps.subs <- subset_samples(acropora, Algal.trt %in% cbn[,i]) # subset the data in the trt for that comparison
  ps.subs.sqrt <- microbiome::transform(ps.subs, "hellinger") 
  metadata_sub <- data.frame(sample_data(ps.subs)) # specify the metadata for that subset
  
  permanova_pairwise <- anosim(phyloseq::distance(ps.subs.sqrt, method = "bray"),
                               metadata_sub$Algal.trt) # perform anosim for the data based on those two treatments
  acropora.bray.p <- c(acropora.bray.p, permanova_pairwise$signif[1]) # append the p value for that comparison to the created p value dataframe
}

acropora.bray.p.adj <- p.adjust(acropora.bray.p, method = "bonferroni") # adjust the p value for multiple comparisons
acropora.bray.p.table <- as.data.frame(cbind.data.frame(t(cbn), p=acropora.bray.p, p.adj=acropora.bray.p.adj)) # create data table containing p value comparisons 
# SN sig different from amansia, empty, TM. SP sig different from empty, TM
# write to csv for table s3

acropora.adonis.p <- c() # Create vector for outputting P values

for(i in 1:ncol(cbn)){ # for each possible comparison
  ps.subs <- subset_samples(acropora, Algal.trt %in% cbn[,i]) # subset the data in the trt for that comparison
  ps.subs.sqrt <- microbiome::transform(ps.subs, "hellinger") 
  metadata_sub <- data.frame(sample_data(ps.subs)) # specify the metadata for that subset
  
  permanova_pairwise <- adonis2(phyloseq::distance(ps.subs.sqrt, method = "bray") ~
                               metadata_sub$Algal.trt) # perform anosim for the data based on those two treatments
  acropora.adonis.p <- c(acropora.adonis.p, permanova_pairwise$`Pr(>F)`[1]) # append the p value for that comparison to the created p value dataframe
}

acropora.adonis.p.adj <- p.adjust(acropora.adonis.p, method = "bonferroni") # adjust the p value for multiple comparisons
acropora.adonis.p.table <- as.data.frame(cbind.data.frame(t(cbn), p=acropora.adonis.p, p.adj=acropora.adonis.p.adj)) # create data table containing p value comparisons
# write to csv for table s3



acropora_bray <- ordinate(acropora_sqrt, distance = "bray", method = "PCoA")




# Plot ----
(acropora_bray_plot <- plot_ordination(physeq = acropora, ordination = acropora_bray, type = "samples", color = "Algal.trt")  + 
  theme_bw() +
  labs(title = "a. *A. pulchra* Bray-Curtis\n", color = "Algal treatment") +
  geom_point(size = 4) +
  geom_text(aes(x = -0.8, y = 0.5, label = "PERMANOVA\nR2 = 0.13, P = 0.001"), colour = 'black', hjust = 0, size = 10) + 

#  coord_fixed() +
  scale_color_manual(values = c(Empty = "grey",
               `Turf mimic` = "black",
               `Macroalgae mimic` = "forestgreen",
               Amansia = "red",
               Sargassum = "brown",
               `S nigricans` = "blue",
               `S punctatus` = "skyblue"),
               breaks = c("Empty", "Turf mimic", "Macroalgae mimic", "Amansia", "Sargassum", "S nigricans", "S punctatus"),
               labels = c("Empty", "Turf mimic", "Macroalgal mimic", expression(italic("Amansia")), expression(italic("Sargassum")),
                          expression(italic("S. nigricans")), expression(italic("S. punctatus" )) ) 
               ) +
  theme(panel.background = element_blank(), 
        panel.grid.major = element_blank(),  #remove major-grid labels
        panel.grid.minor = element_blank(),  #remove minor-grid labels
        plot.background = element_blank(),
                plot.title = element_markdown(color = "grey20", size = 32, hjust = 0, vjust = 0, face = "plain"),
        axis.text.x = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        legend.position = "bottom",
        legend.key.size = unit(2.2, "cm"), # increase legend size
        legend.text = element_text(size = 28), # increase legend text
        legend.title = element_text(size = 30), 
        legend.text.align = 0)  +
  stat_ellipse(aes(group = Algal.trt), linetype = 1) )

#Dispersion test
acropora_dispr <- vegan::betadisper(acropora_bray_matrix, phyloseq::sample_data(acropora)$Algal.trt)
acropora_dispr # 

# plot of dispersion test
plot(acropora_dispr, main = "Ordination Centroids and Dispersion Labeled: Aitchison Distance", sub = "")
boxplot(acropora_dispr, main = "", xlab = "") # turf trts generally furthest from centroid
vegan::permutest(acropora_dispr) # F = 1.8, P = 0.1




## Tukey's Honest Significant Differences
(mod.HSD <- TukeyHSD(acropora_dispr)) # no sig differences


```

```{r porites bray curtis}

# repeat for porites
porites_sqrt <- microbiome::transform(porites_rel_abund, "hellinger") 

porites_bray_matrix <- phyloseq::distance(porites_sqrt, method = "bray") 

anosim(porites_bray_matrix, phyloseq::sample_data(porites_sqrt)$Algal.trt) # R = 0.22, P = 0.001
adonis2(porites_bray_matrix ~ phyloseq::sample_data(porites_sqrt)$Algal.trt) # R2 = 0.13, F = 1.9, P = 0.001

# PAIRWISE COMPARISONS FOR ANOSIM (https://www.yanh.org/2021/01/01/microbiome-r/)

cbn <- combn(x=unique(phyloseq::sample_data(porites)$Algal.trt), m = 2) # Create dataframe specifying possible comparisons


porites.bray.p <- c() # Create vector for outputting P values

for(i in 1:ncol(cbn)){ # for each possible comparison
  ps.subs <- subset_samples(porites, Algal.trt %in% cbn[,i]) # subset the data in the trt for that comparison
  ps.subs.sqrt <- microbiome::transform(ps.subs, "hellinger") 
  metadata_sub <- data.frame(sample_data(ps.subs)) # specify the metadata for that subset
  
  permanova_pairwise <- anosim(phyloseq::distance(ps.subs.sqrt, method = "bray"),
                               metadata_sub$Algal.trt) # perform anosim for the data based on those two treatments
  porites.bray.p <- c(porites.bray.p, permanova_pairwise$signif[1]) # append the p value for that comparison to the created p value dataframe
}

porites.bray.p.adj <- p.adjust(porites.bray.p, method = "bonferroni") # adjust the p value for multiple comparisons
porites.bray.p.table <- as.data.frame(cbind.data.frame(t(cbn), p=porites.bray.p, p.adj=porites.bray.p.adj)) # create data table containing p value comparisons 
# SN sig different from amansia, empty, TM. SP sig different from empty and TM. Sarg also different from empty
# write to csv for table s3


porites.adonis.p <- c() # Create vector for outputting P values

for(i in 1:ncol(cbn)){ # for each possible comparison
  ps.subs <- subset_samples(porites, Algal.trt %in% cbn[,i]) # subset the data in the trt for that comparison
  ps.subs.sqrt <- microbiome::transform(ps.subs, "hellinger") 
  metadata_sub <- data.frame(sample_data(ps.subs)) # specify the metadata for that subset
  
  permanova_pairwise <- adonis2(phyloseq::distance(ps.subs.sqrt, method = "bray") ~
                               metadata_sub$Algal.trt) # perform anosim for the data based on those two treatments
  porites.adonis.p <- c(porites.adonis.p, permanova_pairwise$`Pr(>F)`[1]) # append the p value for that comparison to the created p value dataframe
}

porites.adonis.p.adj <- p.adjust(porites.adonis.p, method = "bonferroni") # adjust the p value for multiple comparisons
porites.adonis.p.table <- as.data.frame(cbind.data.frame(t(cbn), p=porites.adonis.p, p.adj=porites.adonis.p.adj)) # create data table containing p value comparisons
# write to csv for Table S3

porites_bray <- ordinate(porites_sqrt, distance = "bray", method = "PCoA")




(porites_bray_plot <- plot_ordination(physeq = porites, ordination = porites_bray, type = "samples", color = "Algal.trt")  + 
  theme_bw() +
  labs(title = "b. *P. rus* Bray-Curtis\n", color = "Algal treatment") +
  geom_text(aes(x = 0.93, y = 0.55, label = "PERMANOVA\nR2 = 0.13, P = 0.001"), hjust = 1, size = 10, colour = 'black')+
  geom_point(size = 4) +
#  coord_fixed() +
  scale_color_manual(values = c(Empty = "grey",
               `Turf mimic` = "black",
               `Macroalgae mimic` = "forestgreen",
               Amansia = "red",
               Sargassum = "brown",
               `S nigricans` = "blue",
               `S punctatus` = "skyblue"),
               breaks = c("Empty", "Turf mimic", "Macroalgae mimic", "Amansia", "Sargassum", "S nigricans", "S punctatus"),
               labels = c("Empty", "Turf mimic", "Macroalgal mimic", expression(italic("Amansia")), expression(italic("Sargassum")),
                          expression(italic("S. nigricans")), expression(italic("S. punctatus" )) ) 
               ) +
  theme(panel.background = element_blank(), 
        panel.grid.major = element_blank(),  #remove major-grid labels
        panel.grid.minor = element_blank(),  #remove minor-grid labels
        plot.background = element_blank(),
                plot.title = element_markdown(color = "grey20", size = 32, hjust = 0, vjust = 0, face = "plain"),
        axis.text.x = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        axis.text.y = element_text(color = "grey20", size = 20, hjust = .5, vjust = .5, face = "plain"),
        axis.title.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        strip.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
        legend.key.size = unit(4, "cm"), # increase legend size
        legend.position = "top",
        legend.text = element_text(size = 30), # increase legend text
        legend.title = element_text(size = 34) )+
  stat_ellipse(aes(group = Algal.trt), linetype = 1, type = "norm") )

#Dispersion test
porites_dispr <- vegan::betadisper(porites_bray_matrix, phyloseq::sample_data(porites)$Algal.trt)
porites_dispr # 

# plot of dispersion test
plot(porites_dispr, main = "Ordination Centroids and Dispersion Labeled: Aitchison Distance", sub = "")
boxplot(porites_dispr, main = "", xlab = "") # turf trts generally furthest from centroid
vegan::permutest(porites_dispr) # F = 0.8, P = 0.6




## Tukey's Honest Significant Differences
(mod.HSD <- TukeyHSD(porites_dispr)) # no sig differences


```

### Plot - compile figure 5 panels
```{r microbial diversity panel graph, fig.height= 30, fig.width=20}
ggarrange(acropora_bray_plot, 
          porites_bray_plot, 
          acropora_observed_ggplot + theme(axis.text.x=element_blank()),
          porites_observed_ggplot + theme(axis.text.x=element_blank()),
          acropora_adiv_ggplot,
          porites_adiv_ggplot,
          ncol=2, nrow=3, heights = c(1,1,1), widths = c(1,1), common.legend = TRUE,
          legend="top", legend.grob = get_legend(acropora_bray_plot)) 


```

### Figure S3 - Panel graph

```{r jaccard diversity panel graph, fig.height=10, fig.width=20}
ggarrange(acropora_jaccard_plot, 
          porites_jaccard_plot, ncol=2, nrow=1, widths = c(1,1), common.legend = TRUE,
          legend="bottom", legend.grob = get_legend(acropora_bray_plot)) 
```



## Figure S4 - DeSeq Heatmap

```{r acropora deseq summary matrix}
acropora_deseq_summary <- read.csv(here::here("data", "acropora_deseq_summary_matrix.csv"), row.names = 1)
acropora_deseq_summary <- as.matrix(acropora_deseq_summary)
colnames(acropora_deseq_summary) <- rownames(acropora_deseq_summary)
melted_acropora_deseq <- melt(acropora_deseq_summary, na.rm = TRUE)

melted_acropora_deseq <- melted_acropora_deseq %>%
                                mutate(value = ifelse(Var1 == Var2, NA, value)) # replace 0 with NA for matching treatments

# Heatmap

(acropora_heatmap <- ggplot(data = melted_acropora_deseq, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 8) +
 scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
   midpoint = 0, limit = c(-30,30), space = "Lab", 
   name="Delta\n(Trt2 - Trt1)") +
  labs(title = "a. *A. pulchra*", x = "Treatment 2", y = "Treatment 1") +
  
  theme_minimal()+ 
theme(
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(0.5, 0.7),
  legend.direction = "horizontal",
  plot.title = element_markdown(color = "grey20", size = 25, hjust = 0, vjust = 0, face = "plain"),
  axis.text.x = element_text(color = "grey20", size = 20, hjust = 1, vjust = 1, angle = 30, face = "plain"),
  axis.title.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
  axis.text.y = element_text(color = "grey20", size = 20, hjust = 1, vjust = .5, face = "plain"),
  axis.title.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
  strip.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
  legend.key.size = unit(1.5, "cm"), # increase legend size
  legend.text = element_text(size = 14), # increase legend text
  legend.title = element_text(size = 24)))+
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = "top", title.hjust = 0.5))



```

```{r porites deseq summary matrix}
porites_deseq_summary <- read.csv(here::here("data", "porites_deseq_summary_matrix.csv"), row.names = 1)
porites_deseq_summary <- as.matrix(porites_deseq_summary)
colnames(porites_deseq_summary) <- rownames(porites_deseq_summary)
melted_porites_deseq <- melt(porites_deseq_summary, na.rm = TRUE)

melted_porites_deseq <- melted_porites_deseq %>%
                                mutate(value = ifelse(Var1 == Var2, NA, value))

# Heatmap

(porites_heatmap <- ggplot(data = melted_porites_deseq, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 geom_text(aes(Var2, Var1, label = value), color = "black", size = 8) +
 scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
   midpoint = 0, limit = c(-65,65), space = "Lab", 
   name="Delta\n(Trt2 - Trt1)") +
  labs(title = "b. *P. rus*", x = "Treatment 2", y = "Treatment 1") +
  
  theme_minimal()+ 
theme(
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(0.5, 0.7),
  legend.direction = "horizontal",
  plot.title = element_markdown(color = "grey20", size = 25, hjust = 0, vjust = 0, face = "plain"),
  axis.text.x = element_text(color = "grey20", size = 20, hjust = 1, vjust = 1, angle = 30, face = "plain"),
  axis.title.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
  axis.text.y = element_text(color = "grey20", size = 20, hjust = 1, vjust = .5, face = "plain"),
  axis.title.y = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
  strip.text.x = element_text(color = "grey20", size = 25, hjust = .5, vjust = 0, face = "plain"),
  legend.key.size = unit(1.5, "cm"), # increase legend size
  legend.text = element_text(size = 18), # increase legend text
  legend.title = element_text(size = 24)))+
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = "top", title.hjust = 0.5))



```
### plot Figure S4 panel graph
```{r heatmap panel, fig.height = 22, fig.width= 10}

grid.arrange(arrangeGrob(acropora_heatmap , porites_heatmap, nrow = 2)
             )

ggarrange(acropora_heatmap, porites_heatmap, ncol=1, nrow=2, heights = c(1,1)) 
```

### Broad summary info from DeSeq for results and discussion
```{r deseq summary stats}
acropora_deseq <- read.csv(here::here("data", "acropora_just_genus_deseq.csv"), stringsAsFactors = T)

length(levels(acropora_deseq$Genus)) # number of unique genera = 26

length(levels(droplevels(acropora_deseq[grep("sulf", acropora_deseq$Genus), ])$Genus)) # number of genera that contain "sulf" = 4
length(levels(droplevels(acropora_deseq[grep("Sulf", acropora_deseq$Genus), ])$Genus)) # number of genera that START with "SULF" = 0
# total of unique genera that involve sulfur cycling = 4 + Hypnocyclicus = 5


porites_deseq <- read.csv(here::here("data", "porites_just_genus_deseq.csv"), stringsAsFactors = T)
length(levels(porites_deseq$Genus)) # number of unique genera = 97
length(levels(droplevels(porites_deseq[grep("sulf", porites_deseq$Genus), ])$Genus)) # number of genera that contain "sulf" = 10
length(levels(droplevels(porites_deseq[grep("Sulf", porites_deseq$Genus), ])$Genus)) # number of genera that START with "SULF" = 2
# Total genera that involve sulfur cycling = 12 + Hypnocyclicus = 13


```

# Revision 2 amended plots

## Figure 5

### Acropora Bray-curtis

```{r acropora bray curtis plot with vectors, fig.height = 10, fig.width=10}
# For Revision 2 - add relevant species scores vectors to PCoA ----

# vector of relevant species scores: Endozoicomonas, Vibrio, and the significant sulfur cyclers
acropora_important_genera <-
  c("Endozoicomonas", "Vibrio", "Hypnocyclicus", "Carboxylicivirga",
    levels(droplevels(acropora_deseq[grep("sulf", acropora_deseq$Genus), ])$Genus))

# Extract scores
acropora_scores <- acropora_bray$vectors[, 1:2]

# extract genera data
acropora_genus_table <- tax_glom(acropora_rel_abund, taxrank = "Genus")

acropora_genera_data <- as.data.frame(t(otu_table(acropora_genus_table))) 

acropora_taxa_names <- as.data.frame(tax_table(acropora_genus_table))$Genus  # Extract Genus names
colnames(acropora_genera_data) <- acropora_taxa_names  # Replace otus as columns with Genus names

acropora_genera_data_subset <- acropora_genera_data[,colnames(acropora_genera_data) %in% acropora_important_genera ]

# Fit mean centroids for genera of interest
acropora_bray_fit <- envfit(acropora_scores, acropora_genera_data_subset, perm = 999)

# extract vectors
acropora_genus_vectors <- scores(acropora_bray_fit, display = "vectors")

# convert to dataframe
acropora_genus_df <- data.frame(Genus = rownames(acropora_genus_vectors),
                       PCoA1 = acropora_genus_vectors[, 1], 
                       PCoA2 = acropora_genus_vectors[, 2])

(acropora_bray_plot_2 <-
  acropora_bray_plot + 
  geom_segment(data = acropora_genus_df, aes(x = 0, y = 0, xend = PCoA1, yend = PCoA2),
                 arrow = arrow(length = unit(0.2, "cm")), color = "black") +
  geom_text(data = acropora_genus_df, aes(x = PCoA1, y = PCoA2, label = Genus), 
            size = 4, color = "black", vjust = -0.5))


```

```{r porites updated bray, fig.height = 10, fig.width=10}
porites_important_genera <-
  c("Endozoicomonas", "Vibrio", "Hypnocyclicus", "Carboxylicivirga",
    levels(droplevels(porites_deseq[grep("sulf", porites_deseq$Genus), ])$Genus),
                      levels(droplevels(porites_deseq[grep("Sulf", porites_deseq$Genus), ])$Genus))

# Extract scores
porites_scores <- porites_bray$vectors[, 1:2]

# extract genera data
porites_genus_table <- tax_glom(porites_rel_abund, taxrank = "Genus")

porites_genera_data <- as.data.frame(t(otu_table(porites_genus_table))) 

porites_taxa_names <- as.data.frame(tax_table(porites_genus_table))$Genus  # Extract Genus names
colnames(porites_genera_data) <- porites_taxa_names  # Replace otus as columns with Genus names

porites_genera_data_subset <- porites_genera_data[,colnames(porites_genera_data) %in% porites_important_genera ]

# Fit mean centroids for genera of interest
porites_bray_fit <- envfit(porites_scores, porites_genera_data_subset, perm = 999)

# extract vectors
porites_genus_vectors <- scores(porites_bray_fit, display = "vectors")

# convert to dataframe
porites_genus_df <- data.frame(Genus = rownames(porites_genus_vectors),
                       PCoA1 = porites_genus_vectors[, 1], 
                       PCoA2 = porites_genus_vectors[, 2])

(porites_bray_plot_2 <-
  porites_bray_plot + 
  geom_segment(data = porites_genus_df, aes(x = 0, y = 0, xend = PCoA1, yend = PCoA2),
                 arrow = arrow(length = unit(0.2, "cm")), color = "black") +
  geom_text(data = porites_genus_df, aes(x = PCoA1, y = PCoA2, label = Genus), 
            size = 4, color = "black", vjust = -0.5))

```
# R2 more detail on taxa
```{r get summary stats for taxa of interest}

acropora_genera_data_subset$Algal.trt <- acropora_adiv$Algal.trt # add treatment


long_acropora_data <- acropora_genera_data_subset %>%
  pivot_longer(cols = -Algal.trt,  # Keep "Treatment" column intact, pivot everything else
               names_to = "Genus", 
               values_to = "Abundance")

acropora_genus_summary <- long_acropora_data %>%
  group_by(Genus, Algal.trt) %>%
  summarise(
    Mean = mean(Abundance, na.rm = TRUE),  # Mean relative abundance
    SE = sd(Abundance, na.rm = TRUE) / sqrt(n()),  # Standard error
    Max =  max(Abundance, na.rm = TRUE),
    Min = min(Abundance, na.rm = TRUE),
    Present = sum(Abundance != 0)/length(Abundance),
    .groups = "drop"  # Remove unnecessary grouping
  )


porites_genera_data_subset$Algal.trt <- porites_adiv$Algal.trt # add treatment


long_porites_data <- porites_genera_data_subset %>%
  pivot_longer(cols = -Algal.trt,  # Keep "Treatment" column intact, pivot everything else
               names_to = "Genus", 
               values_to = "Abundance")

porites_genus_summary <- long_porites_data %>%
  group_by(Genus, Algal.trt) %>%
  summarise(
    Mean = mean(Abundance, na.rm = TRUE),  # Mean relative abundance
    SE = sd(Abundance, na.rm = TRUE) / sqrt(n()),  # Standard error
    Max =  max(Abundance, na.rm = TRUE),
    Min = min(Abundance, na.rm = TRUE),
    .groups = "drop"  # Remove unnecessary grouping
  )
```

# New Figure S4 - class level viz
```{r stacked bar chart}
acropora_class <- read.csv(here::here("data", "acropora_class_level.csv"))
porites_class <- read.csv(here::here("data", "porites_class_level.csv"))

colnames(porites_class)[which(colnames(porites_class) %in% colnames(acropora_class))]
```